===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.sbin/ntpd/server.c,v
retrieving revision 1.21
retrieving revision 1.22
diff -u -r1.21 -r1.22
--- src/usr.sbin/ntpd/server.c	2005/01/28 12:01:32	1.21
+++ src/usr.sbin/ntpd/server.c	2005/05/24 20:10:50	1.22
@@ -1,4 +1,4 @@
-/*	$OpenBSD: server.c,v 1.21 2005/01/28 12:01:32 dtucker Exp $ */
+/*	$OpenBSD: server.c,v 1.22 2005/05/24 20:10:50 henning Exp $ */
 
 /*
  * Copyright (c) 2003, 2004 Henning Brauer <henning@openbsd.org>
@@ -44,8 +44,9 @@
 		for (ifap = ifa; ifap != NULL; ifap = ifap->ifa_next) {
 			sa = ifap->ifa_addr;
 
-			if (sa->sa_family != AF_INET &&
-			    sa->sa_family != AF_INET6)
+			if (sa == NULL ||
+			    (sa->sa_family != AF_INET &&
+			    sa->sa_family != AF_INET6))
 				continue;
 
 			if ((la = calloc(1, sizeof(struct listen_addr))) ==
